src/lang-promql/complete/hybrid.test.ts

```diff
@@ -0,0 +1,134 @@
+      expectedContext: [{ kind: ContextKind.MetricName }, { kind: ContextKind.Function }, { kind: ContextKind.Aggregation }],
+    },
+    {
+      title: 'autocomplete binOp modifier',
+      expr: 'metric_name / ignor',
+      pos: 19,
+      expectedContext: [
+        { kind: ContextKind.MetricName },
+        { kind: ContextKind.Function },
+        { kind: ContextKind.Aggregation },
+        { kind: ContextKind.BinOpModifier },
+      ],
+    },
+    {
+      title: 'starting to autocomplete labelName in aggregate modifier',
+      expr: 'sum by ()',
+      pos: 8, // cursor is between the bracket
+      expectedContext: [{ kind: ContextKind.LabelName }],
+    },
+    {
+      title: 'continue to autocomplete labelName in aggregate modifier',
+      expr: 'sum by (myL)',
+      pos: 11, // cursor is between the bracket after the string myL
+      expectedContext: [{ kind: ContextKind.LabelName }],
+    },
+    {
+      title: 'autocomplete labelName in a list',
+      expr: 'sum by (myLabel1, myLab)',
+      pos: 23, // cursor is between the bracket after the string myLab
+      expectedContext: [{ kind: ContextKind.LabelName }],
+    },
+    {
+      title: 'autocomplete labelName associated to a metric',
+      expr: 'metric_name{}',
+      pos: 12, // cursor is between the bracket
+      expectedContext: [{ kind: ContextKind.LabelName, metricName: 'metric_name' }],
+    },
+    {
+      title: 'autocomplete labelName that defined a metric',
+      expr: '{}',
+      pos: 1, // cursor is between the bracket
+      expectedContext: [{ kind: ContextKind.LabelName, metricName: '' }],
+    },
+    {
+      title: 'continue to autocomplete labelName associated to a metric',
+      expr: 'metric_name{myL}',
+      pos: 15, // cursor is between the bracket after the string myL
+      expectedContext: [{ kind: ContextKind.LabelName, metricName: 'metric_name' }],
+    },
+    {
+      title: 'continue autocomplete labelName that defined a metric',
+      expr: '{myL}',
+      pos: 4, // cursor is between the bracket after the string myL
+      expectedContext: [{ kind: ContextKind.LabelName, metricName: '' }],
+    },
+    {
+      title: 'autocomplete the labelValue with metricName + labelName',
+      expr: 'metric_name{labelName=""}',
+      pos: 23, // cursor is between the quotes
+      expectedContext: [{ kind: ContextKind.LabelValue, metricName: 'metric_name', labelName: 'labelName' }],
+    },
+    {
+      title: 'autocomplete aggregate operation modifier',
+      expr: 'sum() b',
+      pos: 7, // cursor is between the quotes
+      expectedContext: [{ kind: ContextKind.AggregateOpModifier }],
+    },
+    {
+      title: 'autocomplete aggregate operation modifier 2',
+      expr: 'sum b',
+      pos: 5, // cursor is after 'b'
+      expectedContext: [{ kind: ContextKind.AggregateOpModifier }, { kind: ContextKind.BinOp }],
+    },
+    {
+      title: 'autocomplete binOp',
+      expr: 'metric_name unle',
```

Comment:
Could you add a case for `metric_name !<cursor>`? It should autocomplete `!=`, but it doesn't. Similar for the other ops, it only autocompletes when the typed token is already a valid op.

Reply:
(it worked before btw., same for the other comment)

Reply:
Mmm that's weird. Does it still work on the master branch ?
I'm wondering if it's me that breaks everything or it's coming with cmn v0.13.0

Reply:
Yeah, it still seems to work on master!

Reply:
:'( needs more test then hahaha ! Thanks for the check :)

Reply:
yup indeed, this cases were missing. It's covered now